test/unitTests/spec/pharmacy/controllers/dashboardController.spec.js

```diff
@@ -15,15 +15,65 @@ describe('DashboardController', function () {
     $urlRouterProvider.deferIntercept();
   }));
 
-  beforeEach(inject(function (_$controller_, _$q_, _$rootScope_, _patientService_, _visitService_, _$state_) {
+  beforeEach(inject(function (_$controller_, _$q_, _$rootScope_, _patientService_, _visitService_, _$state_,
+                              _authorizationService_) {
     $controller = _$controller_;
     $q = _$q_;
     $rootScope = _$rootScope_;
     patientService = _patientService_;
     visitService = _visitService_;
     $state = _$state_;
+    authorizationService = _authorizationService_;
   }));
 
+  describe('activate', function () {
+
+    beforeEach(function () {
+
+      spyOn(authorizationService, 'hasRole').and.callFake(function () {
+        return $q(function (resolve) {
+          return resolve(true);
+        });
+      });
+
+      spyOn(patientService, 'getPatient').and.callFake(function () {
+        return $q(function (resolve) {
+          return resolve({});
+        });
+      });
+
+    });
+
+    it('should check if user is independent pharmacist', function () {
+
+      var ctrl = $controller('DashboardController');
+
+      expect(authorizationService.hasRole).toHaveBeenCalled();
+
+    });
+
+    it('should set independentPharmacist property', function () {
+
+      var ctrl = $controller('DashboardController');
+
+      $rootScope.$apply();
```

Comment:
Just curious: Why is ```$rootScope.$apply();``` needed here?

Reply:
...and below

Reply:
To get angular to actually resolve the promises.

https://code.angularjs.org/1.5.11/docs/api/ng/service/$q#differences-between-kris-kowal-s-q-and-q